org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/TypeUtils.java

```diff
@@ -0,0 +1,222 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import com.sun.jdi.ClassLoaderReference;
+import com.sun.jdi.ReferenceType;
+import com.sun.jdi.Type;
+import com.sun.jdi.VirtualMachine;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+/**
+ * The type utility for retrieving display name of the type.
+ */
+public final class TypeUtils {
+    static final char ARRAY = '[';
```

Comment:
There is another code focus on variable, which uses this utility: https://github.com/vscjavaci/eclipse.jdt.ls/blob/685a93f91c527665f2b77f998639b0556e0256f0/org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/VariableUtils.java